eng/Common.props

```diff
@@ -6,6 +6,8 @@
     <TargetOsName Condition=" '$(TargetOsName)' == '' AND $([MSBuild]::IsOSPlatform('FreeBSD'))">freebsd</TargetOsName>
     <TargetArchitecture Condition="'$(TargetArchitecture)' == ''">x64</TargetArchitecture>
     <TargetRuntimeIdentifier Condition="'$(TargetRuntimeIdentifier)' == ''">$(TargetOsName)-$(TargetArchitecture)</TargetRuntimeIdentifier>
+    <PortableBuild Condition="'$(PortableBuild)' == ''">true</PortableBuild>
+    <DefaultAppHostRuntimeIdentifier Condition=" '$(PortableBuild)' == 'false' ">$(TargetRuntimeIdentifier)</DefaultAppHostRuntimeIdentifier>
```

Comment:
Unsure what the existing pattern is but usually in the msbuild world when you want to mimic a boolean, you condition on something either being true or not being true. With conditions that assert on `true` and `false` you are allowing the property to have values that aren't covered by these conditions.

TL;DR: I would expect this to be `'$(PortableBuild)' != 'true'`

Reply:
I had first written it like that, and then inverted it in https://github.com/dotnet/aspnetcore/pull/43937/commits/82414df0502229bc5685e048870f7309729af0ef because of the build failures: https://github.com/dotnet/aspnetcore/runs/8326525459.

I rewrote the check that in case `PortableBuild` is not set, the checks default to portable behavior.